rasa/core/events/__init__.py

```diff
@@ -299,6 +301,13 @@ def as_dict(self) -> Dict[Text, Any]:
         )
         return _dict
 
+    def as_dict_core(self) -> Dict[Text, Text]:
+        entities = [entity.get("entity") for entity in self.entities]
+        if self.intent_name:
+            return {TEXT: None, INTENT: self.intent_name, ENTITIES: entities}
+        else:
+            return {TEXT: self.text, INTENT: None, ENTITIES: entities}
```

Comment:
it'll automatically fill `None` because either `self.text` or `self.intent_name` is `None`
```suggestion
        return {TEXT: self.text, INTENT: self.intent_name, ENTITIES: entities}
```

Reply:
Not really, when we load the intent into UserUttered, it will look like `UserUttered(text = greet{'name':'vova'}, intent = greet, entities = {'name', from: 6, to: 12}) `. And then this full text is used to find entities using the given indices. This is why we need this stupid text...

Reply:
do we use this text somewhere?

Reply:
not for e2e, but some entity extractors

Reply:
please add comments